rmw_connext_cpp/src/rmw_guard_condition.cpp

```diff
@@ -21,8 +22,15 @@
 extern "C"
 {
 rmw_guard_condition_t *
-rmw_create_guard_condition()
+rmw_create_guard_condition(rmw_context_t * context)
 {
+  RCUTILS_CHECK_ARGUMENT_FOR_NULL(context, NULL);
```

Comment:
Could we reduce duplicate code by pushing this into `create_guard_condition`? This would require passing the context and expected identifier as arguments.

Reply:
I think so, but in the interest of getting this pr merged, I'd like to do that in a follow up pr. I'll make an issue on merge.